### PR TITLE
HW29

### PR DESCRIPTION
**Pull request by pavlokhomiak**

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: src/main/java/mate/academy/controllers/IndexController.java [hunk 1]
```diff
@@ -3,22 +3,32 @@
...
+    private final UserService userService =
+            (UserService) injector.getInstance(UserService.class);
 
     @Override
     public void init() throws ServletException {
         super.init();
+
+        User admin = new User("admin", "admin", "admin");
+        admin.setRoles(Set.of(Role.of("ADMIN")));
+        userService.create(admin);
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/controllers/IndexController.java]: let's create a separate InjectDataController for this purpose (and for creating products as well)
we should not place business logic in the init() method

File: src/main/java/mate/academy/web/filter/AuthenticationFilter.java [hunk 2]
```diff
@@ -35,7 +35,7 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
         }
 
         Long userId = (Long) req.getSession().getAttribute(USER_ID);
-        if (userId == null || userService.get(userId) == null) {
+        if (userId == null) {
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/web/filter/AuthenticationFilter.java:36]: let's also refactor AuthenticationFilter as proposed in the checklist

File: src/main/java/mate/academy/web/filter/AuthorizationFilter.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse,
+                         FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        String requestedUrl = req.getServletPath();
+
+        if (protectedUrls.get(requestedUrl) == null) {
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/web/filter/AuthorizationFilter.java]: we can also use `containsKey()` here

File: src/main/java/mate/academy/web/filter/AuthorizationFilter.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        protectedUrls.put("/users/all", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/order/all/admin", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/all/admin", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/add", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/order/all", List.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-carts/products/add", List.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-cart/products/get", List.of(Role.RoleName.USER));
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/web/filter/AuthorizationFilter.java:39]: you also need to add
for ADMIN:

DeleteUserController
DeleteOrderController
DeleteProductController

for USER:

RemoveProductFromCartController
CompleteOrderController

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**